hooks/tk-multi-publish2/ingest/collector.py

```diff
@@ -606,11 +622,88 @@ def _collect_manifest_file(self, settings, parent_item, path):
 
         return file_items
 
+    def on_context_changed(self, settings, item):
+        """
+        Callback to update the item on context changes.
+
+        :param dict settings: Configured settings for this collector
+        :param item: The Item instance
+        """
+
+        # get the item fields and everything in place
+        super(IngestCollectorPlugin, self).on_context_changed(settings, item)
+
+        # restore the fields from the manifest file, even though we currently don't allow users to change context
+        # when ingesting using the manifest file.
+        if "manifest_file_fields" in item.properties:
+            self.logger.info(
+                "Re-creating manifest file fields for: %s" % item.name,
+                extra={
+                    "action_show_more_info": {
+                        "label": "Show Info",
+                        "tooltip": "Show more info",
+                        "text": "Manifest fields:\n%s" %
+                                (pprint.pformat(item.properties.manifest_file_fields))
+                    }
+                }
+            )
+            item.properties.fields.update(item.properties.manifest_file_fields)
+
+        # create the defaults in the item again
+        self.create_ingest_default_fields(settings, item)
+
+    # TODO: Cleanup
```

Comment:
I'm failing to see why you need to override the context on the item. The updates you already made to `_get_work_path_template_from_settings()` on the basic collector plugin should already be sufficient to ensure you are getting the correct work_path_template matching against the file path. If not, you could generate your own temporary context, use it to get the associated environment, then resolve yourself in a method override.

Reply:
This code was for the old method of using `on_widget_context_changed` we now validate the context on item when the user hits validate, and block them from ingesting until they pickup a valid task.

Context is not just for resolving work_path_template. It affects the plugin settings as well.

Reply:
What other settings are being improperly evaluated?

Reply:
Settings for the publisher, won't get resolved correctly if we don't update the context on item.
`work_path_template` is the only setting we are resolving with the other method, because the context is not evaluated by that time.

Rest of the publisher settings like `publish_path_template` or `publish_name_template` still resolve using the item's context.